tests/utils.ts

```diff
@@ -106,7 +106,7 @@ export async function openPlot(
   openObject(page, 'plot', name);
 
   if (waitForLoadFinished) {
-    await expect(page.locator('.iris-grid-panel')).toHaveCount(1);
+    await expect(page.locator('.chart-panel-container')).toHaveCount(1);
```

Comment:
I think we should instead be waiting for the loading spinner to disappear, similar to what we do for grids. But also for grids, we should wait for the panel/loading spinner to appear, then wait for loading to complete. What we had before:
```
  // Expect the panel to be open with a loading spinner first
  await expect(page.locator('.iris-chart-panel')).toHaveCount(1);

  // Wait until it's done loading
  await expect(page.locator('.iris-chart-panel .loading-spinner')).toHaveCount(
    0
  );
```
So I'd probably do:
```
  // Expect the panel to be open with a loading spinner first
  await expect(page.locator('.chart-panel-container .loading-spinner')).toHaveCount(1);

  // Wait until it's done loading
  await expect(page.locator('.chart-panel-container .loading-spinner')).toHaveCount(0);
```
Similarly what we had for tables before:
```

  // Wait for the panel to show
  await expect(page.locator('.iris-grid-panel')).toHaveCount(1);

  // Wait until it's done loading
  await expect(page.locator('.iris-grid-panel .loading-spinner')).toHaveCount(
    0
```

Reply:
I've noticed that expecting the loading spinner to appear first can result in some flakiness if the object finishes loading before that part of the code can be executed.